### DISCUSSION TITLE
Proposal: Add support for Length property to SparseStream class and add a new way to create

### DISCUSSION THREAD

**Discussion by xoma-zver** [dotnet/dotNext#253] [Ideas]
**Length Property:**
This proposal suggests adding support for the Length property to the SparseStream class. This would provide a way to retrieve the total length of the stream.

Potential benefits:

- Improved usability for users who need to know the length of the stream
- Better compatibility with other stream classes that support Length property

Potential drawbacks:

- One significant drawback of implementing the Length property is that it would require enumerating all the underlying streams and retrieving their lengths at the time of SparseStream creation. Enumerating all underlying streams to retrieve their lengths at creation time could be costly or impossible, limiting the flexibility of the SparseStream class and potentially leading to performance issues or errors if the underlying streams don't support length retrieval. .

**New way to create:**
The second proposal implies the possibility of creating a SparseStream instance not only using Combine, but also as an extension method of ReadOnlySpan/IEnumerable, which will simplify the creation of combined streams from a set of streams and avoid awkward constructions like:
```csharp
var fileStreams = filePaths.Select(File.OpenRead).Cast<Stream>().ToArray();
await using var stream = fileStreams.First().Combine(fileStreams.AsSpan(1));
```

**Comment by sakno** [comment]
Taken into consideration as a part of #255 .
